testing/python/fixtures.py

```diff
@@ -1308,6 +1308,32 @@ def test2(no_eq):
         result = pytester.runpytest()
         result.stdout.fnmatch_lines(["*4 passed*"])
 
+    def test_optimize_by_reorder_indirect(self, pytester: Pytester) -> None:
```

Comment:
TODO: Test more cases of how users can provide ids, e.g. `ids=`_Sequence of (or generator for) ids_ (with various return types), and the [pytest_make_parametrize_id](https://docs.pytest.org/en/6.2.x/reference.html#pytest.hookspec.pytest_make_parametrize_id) hook.